app/services/attribute_asserter.rb

```diff
@@ -214,6 +217,15 @@ def add_all_emails(attrs)
     }
   end
 
+  def last_email_from_sp
+    return nil unless IdentityConfig.store.feature_select_email_to_share_enabled
+    sp = service_provider.issuer
+    # identity = user.identities.where(service_provider: sp)
+    identity = user.identities.find_by(service_provider: sp)
+    email_id = identity&.email_address_id
+    return user.email_addresses.find(email_id).email if email_id.is_a? Integer
```

Comment:
What happens if I change the email that's selected for a partner, and then delete that email address? I think that may raise an exception here.

We should probably:

- Have some fallback behavior if the email address can't be found
- Separately, we may need to update the email deletion flow to unset any associated identities using that email address

Reply:
Added accommodation for deleting email